src/components/clusterexplorer/explorer.ts

```diff
@@ -0,0 +1,123 @@
+import * as vscode from 'vscode';
+
+import { Kubectl } from '../../kubectl';
+import * as kubectlUtils from '../../kubectlUtils';
+import { Host } from '../../host';
+import { affectsUs } from '../config/config';
+import { ExplorerExtender, ExplorerUICustomizer } from './explorer.extension';
+import * as providerResult from '../../utils/providerresult';
+import { sleep } from '../../sleep';
+import { refreshExplorer } from '../clusterprovider/common/explorer';
+import { ClusterExplorerNode, ClusterExplorerResourceNode } from './node';
+import { MiniKubeContextNode, ContextNode } from './node.context';
+
+export const KUBERNETES_EXPLORER_NODE_CATEGORY = 'kubernetes-explorer-node';
+
+export type KubernetesExplorerNodeType =
+    'error' |
+    'context' |
+    'folder.resource' |
+    'folder.grouping' |
+    'resource' |
+    'configitem' |
+    'helm.release' |
+    'extension';
+
+export function create(kubectl: Kubectl, host: Host): KubernetesExplorer {
+    return new KubernetesExplorer(kubectl, host);
+}
+
+export function isKubernetesExplorerResourceNode(obj: any): obj is ClusterExplorerResourceNode {
+    return obj && obj.nodeCategory === KUBERNETES_EXPLORER_NODE_CATEGORY && obj.nodeType === 'resource';
```

Comment:
String constant (`resource`) duplicated at least twice, extract as a `const` for sanity.

(here and everywhere else this happens)

Reply:
This used to be impossible because constants had type `string` and the discriminator has type `'resource'`.  But I think TypeScript 3.4 now allows string literal constants to have string literal types.  Good call - I will look into this.